ansible/library/bgp_route.py

```diff
@@ -0,0 +1,343 @@
+        Last update: Fri Sep 22 06:33:39 2017
+
+      65200 62011 65501
+      10.0.0.29 from 10.0.0.29 (100.1.0.15)
+         Origin incomplete, localpref 100, valid, external, multipath
+         Last update: Fri Sep 22 06:33:39 2017
+
+      65200 62011 65501
+      10.0.0.5 from 10.0.0.5 (100.1.0.3)
+         Origin incomplete, localpref 100, valid, external, multipath
+         Last update: Fri Sep 22 06:33:39 2017
+
+      65200 62011 65501
+      10.0.0.21 from 10.0.0.21 (100.1.0.11)
+         Origin incomplete, localpref 100, valid, external, multipath
+         Last update: Fri Sep 22 06:33:37 2017
+
+      65200 62011 65501
+      10.0.0.9 from 10.0.0.9 (100.1.0.5)
+         Origin incomplete, localpref 100, valid, external, multipath, best
+         Last update: Fri Sep 22 06:33:37 2017
+
+
+*  vtysh -c "show ip bgp neighbor 10.0.0.35 adv"
+
+    BGP table version is 0, local router ID is 10.1.0.32
+    Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
+                i internal, r RIB-failure, S Stale, R Removed
+    Origin codes: i - IGP, e - EGP, ? - incomplete
+
+        Network          Next Hop            Metric LocPrf Weight Path
+     *> 10.1.0.32/32     10.0.0.34                 0         32768 i
+     *> 100.1.0.3/32     10.0.0.34                              0 65200 i
+     *> 100.1.0.5/32     10.0.0.34                              0 65200 i
+     *> 100.1.0.7/32     10.0.0.34                              0 65200 i
+     *> 100.1.0.9/32     10.0.0.34                              0 65200 i
+     *> 100.1.0.11/32    10.0.0.34                              0 65200 i
+     *> 100.1.0.13/32    10.0.0.34                              0 65200 i
+     *> 100.1.0.15/32    10.0.0.34                              0 65200 i
+     *> 100.1.0.17/32    10.0.0.34                              0 64001 i
+     *> 192.168.10.97/32 10.0.0.34                              0 65200 62011 65507 ?
+     *> 192.168.10.112/32
+                         10.0.0.34                              0 65200 62011 65508 ?
+     *> 192.168.10.113/32
+                         10.0.0.34                              0 65200 62011 65508 ?
+     *> 192.168.10.128/32
+                         10.0.0.34                              0 65200 62011 65509 ?
+     *> 192.168.10.129/32
+                         10.0.0.34                              0 65200 62011 65509 ?
+     *> 192.168.10.144/32
+                         10.0.0.34                              0 65200 62011 65510 ?
+'''
+
+
+### TODO: No show ipv6 route entries parsing option, some show command will expose ipv6 routes, need continue working on ipv6 specific commands###
+
+
+class BgpRoutes(object):
+    '''
+        parsing bgp routing information
+    '''
+    def __init__(self):
+        self.facts = defaultdict(dict)
+        return
+
+    def get_facts(self):
+        return self.facts
+
+    def parse_bgp_route_adv(self, cmd, cmd_result):
```

Comment:
the argument should be neigh, you do not need to give the cmmand and use regex to re-parse it.